src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] lines = readFromFile(fromFileName).split(System.lineSeparator());
+        int supplyCount = countSpecificType(lines, SUPPLY);
+        int byuCount = countSpecificType(lines, BUY);
+        int resultCount = supplyCount - byuCount;
+        writeToFile(toFileName, toCsvFormat(supplyCount, byuCount, resultCount));
```

Comment:
The 'getStatistic' method is doing too much: it's reading from a file, processing data, and writing to another file. To adhere to the principle of dividing program logic, consider breaking this method into smaller ones, each responsible for a single task.